### PR TITLE
Fix Negative Offset in Hexdump Json Output

### PR DESCRIPTION
**Pull request by AandersonL**
**Checklist**

- [ ] Closing issues: #issue
- [ x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

When the user try to get a json output of a negative relative offset in the `px` command, it will always get a fixed `256` data block.

This fix will rely in the `core->num->value` instead the `core->blocksize` when passing it to `r_print_jsondump` making the output correct.

How to reproduce:

`xj -10`

### CODE REVIEW

File: libr/core/cmd_print.c [hunk 1]
```diff
@@ -6541,7 +6541,7 @@ static int cmd_print(void *data, const char *input) {
 		r_cons_break_push (NULL, NULL);
 		switch (input[1]) {
 		case 'j': // "pxj"
-			r_print_jsondump (core->print, core->block, core->blocksize, 8);
+			r_print_jsondump (core->print, core->block, core->num->value, 8);
```
Comment (trufae) [inline libr/core/cmd_print.c:6544]: Can you test this patch? See other cases how it takes the data.
```suggestion
			r_print_jsondump (core->print, core->block, len, 8);
```

also can you add a test to ensure the output is correct for positive and negative offsets, as well as 0

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/cmd_print.c (hunk 1 above)
Comment (trufae) [reply libr/core/cmd_print.c:6544]: In the tests try also bigger output than the blocksize. Something like:

```
pxj 10
pxj -10
b 16
pxj
pxj 32
pxj-32
```

Comment (AandersonL) [reply libr/core/cmd_print.c]:
```
b 16
pxj
pxj 32
pxj-32
```
The following tests produced a wrong output from the negative value, it cames 16 bytes instead 32

Comment (trufae) [reply libr/core/cmd_print.c]: Yep, i can confirm this issue too. But i think it's fixed if you pass buf and len.

Please create a new test by adding those lines in `test/db/cmd/cmd_print` and run it like this `r2r -i test/db/cmd/cmd_print`.

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -6541,7 +6541,7 @@ static int cmd_print(void *data, const char *input) {
 		r_cons_break_push (NULL, NULL);
 		switch (input[1]) {
 		case 'j': // "pxj"
-			r_print_jsondump (core->print, core->block, core->blocksize, 8);
+			r_print_jsondump (core->print, core->block, len, 8);
```
Comment (trufae) [inline libr/core/cmd_print.c:6544]:
```suggestion
			r_print_jsondump (core->print, buf, len, 8);
```

File: libr/core/cmd_print.c (hunk 1 above)
Comment (trufae) [reply libr/core/cmd_print.c]: ping?
